subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -28,6 +28,17 @@ struct getaddrinfo_state {
 	struct zsock_addrinfo *ai_arr;
 };
 
+/* Initialize static fields of addrinfo structure. A macro to let it work
+ * with any sockaddr_* type.
+ */
+#define INIT_ADDRINFO(addrinfo, sockaddr) { \
+		addrinfo->ai_addr = &addrinfo->_ai_addr; \
+		addrinfo->ai_addrlen = sizeof(*sockaddr); \
+		addrinfo->ai_canonname = addrinfo->_ai_canonname; \
+		*addrinfo->_ai_canonname = '\0'; \
```

Comment:
As the addrinfo is an argument to a macro, it is better to say `addrinfo->_ai_canonname[0]`. The best option would be to have `(addrinfo)->...` in all places. This is not really an issue right now, but could be if the macro is used somewhere else too.

Reply:
> it is better to say addrinfo->_ai_canonname[0]

Done.

> The best option would be to have (addrinfo)->... in all places. 

Done.

> This is not really an issue right now, but could be if the macro is used somewhere else too.

Yes, this macro is supposed to be reused in multiple places (eventually). Actually, the reason I submit this patch is because another more useful patch (support for numeric addresses) depends on this macro, which I have happened to introduce before that.